pkgs/applications/misc/corectrl/default.nix

```diff
@@ -0,0 +1,67 @@
+{ stdenv
+, fetchFromGitLab
+, extra-cmake-modules
+, botan2
+, karchive
+, kauth
+, libdrm
+, mesa-demos
+, procps
+, utillinux
+, vulkan-tools
+, qtbase
+, qtcharts
+, qtquickcontrols2
+, qtsvg
+, qttools
+, qtxmlpatterns
+, wrapQtAppsHook
+} :
+
+stdenv.mkDerivation rec{
+  pname = "corectrl";
+  version = "1.1.1";
+
+  src = fetchFromGitLab {
+    owner = "corectrl";
+    repo = "corectrl";
+    rev = "v${version}";
+    sha256 = "sha256-YQDrxPqCa3OzNKd3UiAffqqvOrgbXmDFJGjYPetolyY=";
+  };
+
+  nativeBuildInputs = [
+    extra-cmake-modules
+    wrapQtAppsHook
+  ];
+  buildInputs = [
+    botan2
+    karchive
+    kauth
+    libdrm
+    mesa-demos
+    procps
+    utillinux
+    vulkan-tools
+    qtbase
+    qtcharts
+    qtquickcontrols2
+    qtsvg
+    qttools
+    qtxmlpatterns
+  ];
+
+  meta = with stdenv.lib; {
+    homepage = "https://gitlab.com/corectrl/corectrl/";
+    description = "Control your computer hardware via application profiles";
+    longDescription = ''
+      CoreCtrl is a Free and Open Source GNU/Linux application that allows you
+      to control with ease your computer hardware using application profiles. It
+      aims to be flexible, comfortable and accessible to regular users.
+    '';
+    licenses = licenses.gpl3Plus;
+    platforms = platforms.linux;
+  };
+}
+# TODO: remind upstream that libdrm is not detected at configure time
+# TODO: add vulkaninfo and glxinfo to $PATH
+# TODO: http://ix.io/2AKo
```

Comment:
Can someone help me here, please? I have tried to test it, but the application hangs until be C-c'ed.

```kf.auth: Tried to start an invalid action:  "org.corectrl.helper.init"```

http://ix.io/2AKo

Reply:
I gave a quick try and it seems that corectrl attempts to launch a helper on start (this is the dbus action you see).
Eavesdropping on the system bus (dbus-monitor --system), I've seen the following output:
`"The name org.corectrl.helper was not provided by any .service files"`.

Indeed there is no such service on my current system, but the result contains a definition in `results/corectrl/share/dbus-1/system-services/org.corectrl.helper.service`.

Not sure how to make dbus aware of this, but that's probably the missing part.

Reply:
So the best way is to make something like a NixOS module, maybe?

Reply:
@greizgh I added vulkanInfo and glxinfo stuff